CMakeLists.txt

```diff
@@ -5,7 +5,7 @@
 #
 # Please read The Ultimate Guide to CMake:
 # https://rix0r.nl/blog/2015/08/13/cmake-guide/
-cmake_minimum_required(VERSION 3.5.1 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
```

Comment:
With @dragoshsch's work, we could bump this to 3.13, and I think we might as well so long as CI is updated and we're doing the bump.